Build failure with version 1.9.9 for ESP32-C3 target

I updated ESP8266Audio in my project from 1.9.7 to 1.9.9 and since then, I receive the following errors when trying to compile for the ESP32-C3 target platform (RISC-V). Building for ESP8266 (Xtensa) works.
I don't even use the failing component, my Project only includes the following headers:

```
#include "AudioFileSourceICYStream.h"
#include "AudioFileSourceBuffer.h"
#include "AudioGeneratorMP3.h"
#include "AudioOutputI2S.h"
```

```
    ~/Documents/GitLab/radio-2/firmware    main !2 ?1  pio run --target upload -e esp32-c3-devkit                                                                                                                                                                                                    1 ✘  4s  
Processing esp32-c3-devkit (platform: espressif32; board: esp32-c3-devkitm-1; framework: arduino)
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Removing unused dependencies...
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32-c3-devkitm-1.html
PLATFORM: Espressif 32 (6.9.0) > Espressif ESP32-C3-DevKitM-1
HARDWARE: ESP32C3 160MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-builtin, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES: 
 - framework-arduinoespressif32 @ 3.20017.241212+sha.dcc1105b 
 - tool-esptoolpy @ 1.40501.0 (4.5.1) 
 - tool-mkfatfs @ 2.0.1 
 - tool-mklittlefs @ 1.203.210628 (2.3) 
 - tool-mkspiffs @ 2.230.0 (2.30) 
 - toolchain-riscv32-esp @ 8.4.0+2021r2-patch5
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Library Manager: Installing ESP8266audio @ ^1.9.9
Unpacking 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Library Manager: ESP8266Audio@1.9.9 has been installed!
Found 38 compatible libraries
Scanning dependencies...
Dependency Graph
|-- ESP8266Audio @ 1.9.9
|-- Wire @ 2.0.0
|-- Adafruit GFX Library @ 1.11.11
|-- Adafruit BusIO @ 1.16.2
|-- Adafruit SSD1306 @ 2.5.13
|-- RotaryEncoder @ 1.5.3
|-- SPI @ 2.0.0
|-- WiFi @ 2.0.0
Building in release mode
Compiling .pio/build/esp32-c3-devkit/src/AudioPlayer.cpp.o
Compiling .pio/build/esp32-c3-devkit/src/DisplayRenderer.cpp.o
Compiling .pio/build/esp32-c3-devkit/src/RotaryInput.cpp.o
Compiling .pio/build/esp32-c3-devkit/src/StateMachine.cpp.o
Compiling .pio/build/esp32-c3-devkit/src/WifiManager.cpp.o
Compiling .pio/build/esp32-c3-devkit/src/main.cpp.o
Building .pio/build/esp32-c3-devkit/bootloader.bin
Generating partitions .pio/build/esp32-c3-devkit/partitions.bin
esptool.py v4.5.1
Creating esp32c3 image...
Merged 1 ELF section
Successfully created esp32c3 image.
Compiling .pio/build/esp32-c3-devkit/lib799/FS/FS.cpp.o
Compiling .pio/build/esp32-c3-devkit/lib799/FS/vfs_api.cpp.o
Compiling .pio/build/esp32-c3-devkit/libcd4/SPIFFS/SPIFFS.cpp.o
Compiling .pio/build/esp32-c3-devkit/lib6a0/I2S/I2S.cpp.o
Compiling .pio/build/esp32-c3-devkit/lib10e/WiFi/WiFi.cpp.o
Compiling .pio/build/esp32-c3-devkit/lib10e/WiFi/WiFiAP.cpp.o
Compiling .pio/build/esp32-c3-devkit/lib10e/WiFi/WiFiClient.cpp.o
Compiling .pio/build/esp32-c3-devkit/lib10e/WiFi/WiFiGeneric.cpp.o
Compiling .pio/build/esp32-c3-devkit/lib10e/WiFi/WiFiMulti.cpp.o
Compiling .pio/build/esp32-c3-devkit/lib10e/WiFi/WiFiSTA.cpp.o
Compiling .pio/build/esp32-c3-devkit/lib10e/WiFi/WiFiScan.cpp.o
Compiling .pio/build/esp32-c3-devkit/lib10e/WiFi/WiFiServer.cpp.o
Compiling .pio/build/esp32-c3-devkit/lib10e/WiFi/WiFiUdp.cpp.o
Archiving .pio/build/esp32-c3-devkit/libcd4/libSPIFFS.a
Indexing .pio/build/esp32-c3-devkit/libcd4/libSPIFFS.a
Compiling .pio/build/esp32-c3-devkit/lib7e2/WiFiClientSecure/WiFiClientSecure.cpp.o
Archiving .pio/build/esp32-c3-devkit/lib799/libFS.a
Indexing .pio/build/esp32-c3-devkit/lib799/libFS.a
Compiling .pio/build/esp32-c3-devkit/lib7e2/WiFiClientSecure/esp_crt_bundle.c.o
Compiling .pio/build/esp32-c3-devkit/lib7e2/WiFiClientSecure/ssl_client.cpp.o
Archiving .pio/build/esp32-c3-devkit/lib6a0/libI2S.a
Indexing .pio/build/esp32-c3-devkit/lib6a0/libI2S.a
Compiling .pio/build/esp32-c3-devkit/lib9c3/HTTPClient/HTTPClient.cpp.o
Compiling .pio/build/esp32-c3-devkit/liba4e/SPI/SPI.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf4b/SD/SD.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf4b/SD/sd_diskio.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf4b/SD/sd_diskio_crc.c.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioFileSourceBuffer.cpp.o
Archiving .pio/build/esp32-c3-devkit/lib10e/libWiFi.a
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioFileSourceFS.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioFileSourceFunction.cpp.o
Indexing .pio/build/esp32-c3-devkit/lib10e/libWiFi.a
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioFileSourceHTTPStream.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioFileSourceICYStream.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioFileSourceID3.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioFileSourcePROGMEM.cpp.o
Archiving .pio/build/esp32-c3-devkit/lib7e2/libWiFiClientSecure.a
Indexing .pio/build/esp32-c3-devkit/lib7e2/libWiFiClientSecure.a
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioFileSourceSD.cpp.o
Archiving .pio/build/esp32-c3-devkit/liba4e/libSPI.a
Indexing .pio/build/esp32-c3-devkit/liba4e/libSPI.a
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioFileSourceSPIRAMBuffer.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioFileSourceSTDIO.cpp.o
Archiving .pio/build/esp32-c3-devkit/libf4b/libSD.a
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioFileStream.cpp.o
Indexing .pio/build/esp32-c3-devkit/libf4b/libSD.a
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorAAC.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorFLAC.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorMIDI.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorMOD.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorMP3.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorMP3a.cpp.o
Archiving .pio/build/esp32-c3-devkit/lib9c3/libHTTPClient.a
Indexing .pio/build/esp32-c3-devkit/lib9c3/libHTTPClient.a
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorOpus.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorRTTTL.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorTalkie.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorWAV.cpp.o
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioLogger.cpp.o
In file included from .pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:59:
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:56:5: error: 'tsf' does not name a type
     tsf *g_tsf;
     ^~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:23: error: field 'buffer' has incomplete type 'tsf_stream'
     struct tsf_stream buffer;
                       ^~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;
            ^~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:58:23: error: field 'afsMIDI' has incomplete type 'tsf_stream'
     struct tsf_stream afsMIDI;
                       ^~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;
            ^~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:59:23: error: field 'afsSF2' has incomplete type 'tsf_stream'
     struct tsf_stream afsSF2;
                       ^~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;
            ^~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp: In member function 'void AudioGeneratorMIDI::PrepareMIDI(AudioFileSource*)':
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:342:3: error: 'tsf_stream_wrap_cached' was not declared in this scope
   tsf_stream_wrap_cached(&afsMIDI, 32, 64, &buffer);
   ^~~~~~~~~~~~~~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:342:3: note: suggested alternative: 'esp_spiram_init_cache'
   tsf_stream_wrap_cached(&afsMIDI, 32, 64, &buffer);
   ^~~~~~~~~~~~~~~~~~~~~~
   esp_spiram_init_cache
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp: In member function 'int AudioGeneratorMIDI::PlayMIDI()':
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:446:27: error: 'g_tsf' was not declared in this scope
             tsf_note_off (g_tsf, tg->instrument, tg->note);
                           ^~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:446:27: note: suggested alternative: 'gets'
             tsf_note_off (g_tsf, tg->instrument, tg->note);
                           ^~~~~
                           gets
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:446:13: error: 'tsf_note_off' was not declared in this scope
             tsf_note_off (g_tsf, tg->instrument, tg->note);
             ^~~~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:479:22: error: 'g_tsf' was not declared in this scope
         tsf_note_on (g_tsf, tg->instrument, tg->note, trk->velocity / 127.0); // velocity = 0...127
                      ^~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:479:22: note: suggested alternative: 'gets'
         tsf_note_on (g_tsf, tg->instrument, tg->note, trk->velocity / 127.0); // velocity = 0...127
                      ^~~~~
                      gets
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:479:9: error: 'tsf_note_on' was not declared in this scope
         tsf_note_on (g_tsf, tg->instrument, tg->note, trk->velocity / 127.0); // velocity = 0...127
         ^~~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp: In member function 'void AudioGeneratorMIDI::StopMIDI()':
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:495:13: error: 'g_tsf' was not declared in this scope
   tsf_close(g_tsf);
             ^~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:495:13: note: suggested alternative: 'gets'
   tsf_close(g_tsf);
             ^~~~~
             gets
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:495:3: error: 'tsf_close' was not declared in this scope
   tsf_close(g_tsf);
   ^~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:495:3: note: suggested alternative: 'afs_close'
   tsf_close(g_tsf);
   ^~~~~~~~~
   afs_close
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp: In member function 'virtual bool AudioGeneratorMIDI::begin(AudioFileSource*, AudioOutput*)':
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:513:3: error: 'g_tsf' was not declared in this scope
   g_tsf = tsf_load(&afsSF2);
   ^~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:513:3: note: suggested alternative: 'gets'
   g_tsf = tsf_load(&afsSF2);
   ^~~~~
   gets
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:513:11: error: 'tsf_load' was not declared in this scope
   g_tsf = tsf_load(&afsSF2);
           ^~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:513:11: note: suggested alternative: 'isr_log_d'
   g_tsf = tsf_load(&afsSF2);
           ^~~~~~~~
           isr_log_d
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:515:26: error: 'TSF_MONO' was not declared in this scope
   tsf_set_output (g_tsf, TSF_MONO, freq, -10 /* dB gain -10 */ );
                          ^~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:515:26: note: suggested alternative: 'DST_NONE'
   tsf_set_output (g_tsf, TSF_MONO, freq, -10 /* dB gain -10 */ );
                          ^~~~~~~~
                          DST_NONE
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:515:3: error: 'tsf_set_output' was not declared in this scope
   tsf_set_output (g_tsf, TSF_MONO, freq, -10 /* dB gain -10 */ );
   ^~~~~~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp: In member function 'virtual bool AudioGeneratorMIDI::loop()':
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:561:29: error: 'g_tsf' was not declared in this scope
       tsf_render_short_fast(g_tsf, samplesRendered, numSamplesRendered, 0);
                             ^~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:561:29: note: suggested alternative: 'gets'
       tsf_render_short_fast(g_tsf, samplesRendered, numSamplesRendered, 0);
                             ^~~~~
                             gets
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:561:7: error: 'tsf_render_short_fast' was not declared in this scope
       tsf_render_short_fast(g_tsf, samplesRendered, numSamplesRendered, 0);
       ^~~~~~~~~~~~~~~~~~~~~
Compiling .pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioOutputBuffer.cpp.o
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp: In member function 'void AudioGeneratorMIDI::MakeStreamFromAFS(AudioFileSource*, tsf_stream*)':
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:635:6: error: invalid use of incomplete type 'struct tsf_stream'
   afs->data = reinterpret_cast<void*>(src);
      ^~
In file included from .pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:59:
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;
            ^~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:636:6: error: invalid use of incomplete type 'struct tsf_stream'
   afs->read = &afs_read;
      ^~
In file included from .pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:59:
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;
            ^~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:637:6: error: invalid use of incomplete type 'struct tsf_stream'
   afs->tell = &afs_tell;
      ^~
In file included from .pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:59:
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;
            ^~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:638:6: error: invalid use of incomplete type 'struct tsf_stream'
   afs->skip = &afs_skip;
      ^~
In file included from .pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:59:
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;
            ^~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:639:6: error: invalid use of incomplete type 'struct tsf_stream'
   afs->seek = &afs_seek;
      ^~
In file included from .pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:59:
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;
            ^~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:640:6: error: invalid use of incomplete type 'struct tsf_stream'
   afs->close = &afs_close;
      ^~
In file included from .pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:59:
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;
            ^~~~~~~~~~
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:641:6: error: invalid use of incomplete type 'struct tsf_stream'
   afs->size = &afs_size;
      ^~
In file included from .pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:59:
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;
            ^~~~~~~~~~
*** [.pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorMIDI.cpp.o] Error 1
============================================================================================================================================= [FAILED] Took 8.83 seconds =============================================================================================================================================

Environment      Status    Duration
---------------  --------  ------------
esp32-c3-devkit  FAILED    00:00:08.829
======================================================================================================================================= 1 failed, 0 succeeded in 00:00:08.829 =======================================================================================================================================
    ~/Documents/GitLab/radio-2/firmware    main !2 ?1  >....                                                                                                                                                                                                                                                     
            ^~~~~~~~~~                                                                        
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:640:6: error: invalid use of incomplete type 'struct tsf_stream'
   afs->close = &afs_close;                                                                                                          
      ^~                   
In file included from .pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:59:                                       
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;                                                                                            
            ^~~~~~~~~~                                                                        
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:641:6: error: invalid use of incomplete type 'struct tsf_stream'
   afs->size = &afs_size;                                                                                                            
      ^~                 
In file included from .pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.cpp:59:                                       
.pio/libdeps/esp32-c3-devkit/ESP8266Audio/src/AudioGeneratorMIDI.h:57:12: note: forward declaration of 'struct tsf_stream'
     struct tsf_stream buffer;                                                                                            
            ^~~~~~~~~~                                                                        
*** [.pio/build/esp32-c3-devkit/libf0a/ESP8266Audio/AudioGeneratorMIDI.cpp.o] Error 1                                     
============================================================================================================================================= [FAILED] Took 8.83 seconds ============================================================================================================================================= 
 
Environment      Status    Duration
---------------  --------  ------------
esp32-c3-devkit  FAILED    00:00:08.829
======================================================================================================================================= 1 failed, 0 succeeded in 00:00:08.829 =======================================================================================================================================
```

Error happens on Fedora 41 and macOS 15.2 (Intel), so I assume the build platform is not relevant.

--- a/src/libtinysoundfont/tsf.h
+++ b/src/libtinysoundfont/tsf.h
@@ -50,7 +50,9 @@
 //  2101 | }
 //       | ^
 
-#if !defined(ESP32)
+#if defined(ESP32) && (__GNUC__ >= 8) && (__XTENSA__)
+// Not defined, can't build
+#else
 
 #ifndef TSF_INCLUDE_TSF_INL
 #define TSF_INCLUDE_TSF_INL

